src/ElasticSearchIndexer.php

```diff
@@ -29,115 +32,101 @@
+    public function __construct(Client $client)
     {
         $this->client = $client;
-        $this->index = $index;
-        $this->type = $type;
     }
 
-    /**
-     * @return string
-     */
-    public function getIndex()
+    protected function getClient(): Client
     {
-        return $this->index;
+        return $this->client;
     }
 
-    /**
-     * @return string
-     */
-    public function getType()
+    public function getIndexNameByDocument(IndexDocument $document): string
     {
-        return $this->type;
-    }
+        $documentBody = $document->getBody();
 
-    /**
-     * @return Client|null
-     */
-    protected function getClient()
-    {
-        return $this->client;
+        if (!isset($documentBody['type'])) {
+            throw new RuntimeException('type property is undefined on the document');
+        }
+
+        $documentType = $documentBody['type'];
+
+        foreach (self::AVAILABLE_INDEXES as $ontology => $indexName) {
+            if (in_array($ontology, $documentType)) {
+                return $indexName;
+            }
+        }
+
+        return self::UNCLASSIFIEDS_DOCUMENTS_INDEX;
     }
 
     /**
      * @param Iterator $documents
-     * @return int
+     * @return int The number of indexed documents
      */
-    public function buildIndex(Iterator $documents)
+    public function buildIndex(Iterator $documents): int
     {
         $count = 0;
+        $blockSize = 0;
+        $params = [];
+
         while ($documents->valid()) {
-            $blockSize = 0;
-            $params = ['body' => []];
-
-            while ($documents->valid() && $blockSize < self::INDEXING_BLOCK_SIZE) {
-                /** @var IndexDocument $document */
-                $document = $documents->current();
-
-                // First step we trying to create document. If is exist, then skip this step
-                $params['body'][] = [
-                    'create' => [
-                        '_index' => $this->getIndex(),
-                        '_type' => $this->getType(),
-                        '_id' => $document->getId()
-                    ]
-                ];
+            /** @var IndexDocument $document */
+            $document = $documents->current();
 
-                $params['body'][] = $document->getBody();
+            // First step we trying to create document. If is exist, then skip this step
+            $indexName = $this->getIndexNameByDocument($document);
 
-                // Trying to update document
-                $params['body'][] = [
-                    'update' => [
-                        '_index' => $this->getIndex(),
-                        '_type' => $this->getType(),
-                        '_id' => $document->getId()
-                    ]
-                ];
+            if ($indexName === self::UNCLASSIFIEDS_DOCUMENTS_INDEX) {
+                \common_Logger::i(sprintf('There is no proper index for the document "%s"', $document->getId()));
```

Comment:
Maybe inject the logger instead of using these static calls?

Reply:
Thank you Luis, fixed.